pdc/apps/common/viewsets.py

```diff
@@ -63,6 +65,64 @@ def perform_create(self, serializer):
                                        _dumps_json(item.export()))
 
 
+def get_related_attr(obj, spec):
+    fields = spec.split('__')
+    for f in fields:
+        obj = getattr(obj, f)
+    return obj
+
+
+class NotificationMixin(object):
+    def get_object(self):
+        obj = super(NotificationMixin, self).get_object()
+        serializer_class = self.get_serializer_class()
+        self.initial_status = serializer_class(instance=obj,
+                                               context={'request': self.request}).data
+        return obj
+
+    def _get_url(self, basename):
+        if not hasattr(self, 'object') or not self.object:
+            return None
+        if hasattr(self, 'lookup_fields'):
+            lookup_field_values = [get_related_attr(self.object, f) for f, _ in self.lookup_fields]
+            lookup_field = '/'.join(lookup_field_values)
```

Comment:
Won't work for `releases/{release_id}/rpm-mapping/{package}/`.

This is bit crazy, but could help:
```python
from django.core.urlresolvers import get_resolver
resolver = get_resolver(None)
resolver.reverse_dict.getlist('releaserpmmapping-detail')[1][0][0] # This is bit random :)
# (u'rest_api/v1/releases/%(release_id)s/rpm-mapping/%(package)s/', [u'release_id', u'package'])
```

No idea if there is simple way to get URL. :(